File: src/types/prompt.ts
// types/prompt.ts — 与 src-tauri/src/models/prompt.rs 一一对应。
export interface Prompt {
  id: number;
  title: string;
  content: string;
  folder_id: number | null;
  tag_ids: number[];
  is_favorite: boolean;
  is_pinned: boolean;
  use_count: number;
  last_used_at: number | null; // unix ms
  created_at: number;
  updated_at: number;
}

export interface PromptDraft {
  title: string;
  content: string;
  folder_id: number | null;
  tag_ids: number[];
}

export type SortMode =
  | "recent_used"
  | "created"
  | "updated"
  | "title";

File: src-tauri/src/models/prompt.rs
// models/prompt.rs — 镜像 src/types/prompt.ts。
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct Prompt {
    pub id: i64,
    pub title: String,
    pub content: String,
    pub folder_id: Option<i64>,
    pub tag_ids: Vec<i64>,
    pub is_favorite: bool,
    pub is_pinned: bool,
    pub use_count: i64,
    pub last_used_at: Option<i64>,
    pub created_at: i64,
    pub updated_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct PromptDraft {
    pub title: String,
    pub content: String,
    pub folder_id: Option<i64>,
    pub tag_ids: Vec<i64>,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum SortMode {
    RecentUsed,
    Created,
    Updated,
    Title,
}
